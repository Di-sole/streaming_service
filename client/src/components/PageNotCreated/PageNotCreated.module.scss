.container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: 30px 0px;

    & img {
        margin-bottom: 10px;
    }
    & p {
        color: rgba(255, 255, 255, 0.48);
        font-size: 20px;
        text-align: center;
    }
}

.button {
    margin-top: 20px;
}
