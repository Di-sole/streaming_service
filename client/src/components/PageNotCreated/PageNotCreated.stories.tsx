import { Meta, StoryObj } from '@storybook/react';
import './PageNotCreated.module.scss';
import PageNot from './PageNotCreated';

const meta: Meta<typeof PageNot> = {
    title: 'Information/PageNotCreated',
    component: PageNot,
};

export default meta;

type Story = StoryObj<typeof PageNot>;

export const PageNotCreated: Story = {};
