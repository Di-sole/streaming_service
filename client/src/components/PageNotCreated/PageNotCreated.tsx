import { useRouter } from 'next/router';
import BorderedButton from '../UI/buttons/BorderedButton/BorderedButton';
import styles from './PageNotCreated.module.scss';
import { useTranslation } from 'next-i18next';
import attentionIcon from '../../../public/icons/attention.svg';
import Image from 'next/image';

interface PageNotCreatedProps {
    showButton?: boolean;
}

const PageNotCreated = ({ showButton = true }: PageNotCreatedProps) => {
    const router = useRouter();
    const { t } = useTranslation('moviesPage');
    return (
        <div className={styles.container} data-testid={'pageNotCreated'}>
            <Image src={attentionIcon} alt="Внимание" />
            <p>{t('pageError')}</p>
            {showButton && (
                <BorderedButton size="medium" className={styles.button} onClick={() => router.back()}>
                    {t('buttonBack')}
                </BorderedButton>
            )}
        </div>
    );
};

export default PageNotCreated;
