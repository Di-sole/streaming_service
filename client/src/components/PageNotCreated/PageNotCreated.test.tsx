import { render, screen } from '@testing-library/react';
import PageNotCreated from './PageNotCreated';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

jest.mock('next/router', () => ({
    useRouter() {
        return {
            pathname: '',
        };
    },
}));

describe('PAGENOTCREATED SNAPSHOTS TESTS', () => {
    test('PageNotCreated should not be changed', () => {
        render(<PageNotCreated />);

        const id = screen.getByTestId('pageNotCreated');
        expect(id).toMatchSnapshot();
    });
});
