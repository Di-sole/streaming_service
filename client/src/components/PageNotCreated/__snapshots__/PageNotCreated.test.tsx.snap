// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PAGENOTCREATED SNAPSHOTS TESTS PageNotCreated should not be changed 1`] = `
<div
  class="container"
  data-testid="pageNotCreated"
>
  <img
    alt="Внимание"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style="color: transparent;"
    width="40"
  />
  <p>
    pageError
  </p>
  <button
    class="button button_medium button"
    type="button"
  >
    buttonBack
  </button>
</div>
`;
